src/field/pokemon.ts

```diff
@@ -4044,6 +4064,47 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return true;
   }
 
+  protected setStatus(
+    effect: StatusEffect,
+    { toxicTurnCount = 0, sleepTurnsRemaining = 0 }: { toxicTurnCount?: number; sleepTurnsRemaining?: number },
+  ): void {
+    this.status = {
+      effect,
+      toxicTurnCount,
+      sleepTurnsRemaining,
+    };
+  }
+
+  public advanceStatusCounter(): void {
+    if (!this.status) {
+      return;
+    }
+    switch (this.status.effect) {
+      case StatusEffect.TOXIC:
+        this.status.toxicTurnCount++;
+        break;
+      case StatusEffect.SLEEP:
+        if (Overrides.STATUS_ACTIVATION_OVERRIDE === true) {
+          this.status.sleepTurnsRemaining = Math.max(this.status.sleepTurnsRemaining, 1);
+          break;
+        }
+        if (Overrides.STATUS_ACTIVATION_OVERRIDE === false) {
+          this.status.sleepTurnsRemaining = 0;
+          break;
+        }
+        this.status.sleepTurnsRemaining--;
+        break;
+    }
+  }
```

Comment:
NIT: It's generally recommended to always have a `default` block in a switch-case.
```suggestion
    switch (this.status.effect) {
      case StatusEffect.TOXIC:
        this.status.toxicTurnCount++;
        break;
      case StatusEffect.SLEEP:
        if (Overrides.STATUS_ACTIVATION_OVERRIDE === true) {
          this.status.sleepTurnsRemaining = Math.max(this.status.sleepTurnsRemaining, 1);
          break;
        }
        if (Overrides.STATUS_ACTIVATION_OVERRIDE === false) {
          this.status.sleepTurnsRemaining = 0;
          break;
        }
        this.status.sleepTurnsRemaining--;
        break;
      default:
        break;
    }
  }
```